layers/+os/nixos/layers.el

```diff
@@ -0,0 +1,12 @@
+;;; layers.el --- NixOS layers File for Spacemacs
+;;
+;; Copyright (c) 2015-2018 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(configuration-layer/declare-layer-dependencies '(auto-completion))
```

Comment:
This should not be necessary, normally users can disable auto completion as well as syntax checking globally by not loading the respective configuration layers. In your case you are enforcing `auto-completion` globally for all layers when `nixos layer` is loaded.

Please remove the layer reference. In the case of it not being present post init for company and company-nixos-options will not be loaded, so this should not require changes in the package.el code.